src/key.rs

```diff
@@ -595,6 +595,7 @@ impl Connection {
         match self.cached_metadata.as_ref().and_then(|m| m.pin_policy) {
             Some(PinPolicy::Never) => return Ok(Ok(())),
             Some(PinPolicy::Once) if self.yubikey.verify_pin(&[]).is_ok() => return Ok(Ok(())),
+            Some(PinPolicy::Always) if self.yubikey.verify_pin(&[]).is_ok() => return Ok(Ok(())),
```

Comment:
Can you explain how this resolves #80? It appears that this is checking whether we can skip the PIN for `PinPolicy::Always`, but we should never be able to skip the PIN in that situation?